capa/features/extractors/dnfile/insn.py

```diff
@@ -8,82 +8,118 @@
-            ctx["methods_cache"][token] = name
+        for method in get_dotnet_managed_methods(ctx["pe"]):
+            ctx["methods_cache"][method.token] = method
     return ctx["methods_cache"]
 
 
-def get_callee_name(ctx: Dict, token: int) -> str:
-    """map dotnet token to method name"""
-    name: str = get_managed_imports(ctx).get(token, "")
-    if not name:
+def get_callee(ctx: Dict, token: int) -> Union[DnMethod, DnUnmanagedMethod, None]:
+    """map dotnet token to un/managed method"""
+    callee: Union[DnMethod, DnUnmanagedMethod, None] = get_managed_imports(ctx).get(token, None)
+    if not callee:
         # we must check unmanaged imports before managed methods because we map forwarded managed methods
         # to their unmanaged imports; we prefer a forwarded managed method be mapped to its unmanaged import for analysis
-        name = get_unmanaged_imports(ctx).get(token, "")
-        if not name:
-            name = get_methods(ctx).get(token, "")
-    return name
+        callee = get_unmanaged_imports(ctx).get(token, None)
+        if not callee:
+            callee = get_methods(ctx).get(token, None)
+    return callee
 
 
 def extract_insn_api_features(f: CilMethodBody, bb: CilMethodBody, insn: Instruction) -> Iterator[Tuple[API, int]]:
     """parse instruction API features"""
     if insn.opcode not in (OpCodes.Call, OpCodes.Callvirt, OpCodes.Jmp, OpCodes.Calli):
         return
 
-    name: str = get_callee_name(f.ctx, insn.operand.value)
-    if not name:
+    callee: Union[DnMethod, DnUnmanagedMethod, None] = get_callee(f.ctx, insn.operand.value)
+    if callee is None:
         return
 
-    if "::" in name:
-        # like System.IO.File::OpenRead
-        yield API(name), insn.offset
-    else:
+    if isinstance(callee, DnUnmanagedMethod):
         # like kernel32.CreateFileA
-        dll, _, symbol = name.rpartition(".")
-        for name_variant in capa.features.extractors.helpers.generate_symbols(dll, symbol):
-            yield API(name_variant), insn.offset
+        for name in capa.features.extractors.helpers.generate_symbols(callee.modulename, callee.methodname):
+            yield API(name), insn.offset
+    else:
+        # like System.IO.File::Delete
+        yield API(str(callee)), insn.offset
+
+
+def extract_insn_class_features(f: CilMethodBody, bb: CilMethodBody, insn: Instruction) -> Iterator[Tuple[Class, int]]:
+    """parse instruction class features"""
+    if insn.opcode not in (OpCodes.Call, OpCodes.Callvirt, OpCodes.Jmp, OpCodes.Calli):
+        return
+
+    row: Any = resolve_dotnet_token(f.ctx["pe"], Token(insn.operand.value))
+
+    if not isinstance(row, dnfile.mdtable.MemberRefRow):
+        return
+    if not isinstance(row.Class.row, (dnfile.mdtable.TypeRefRow, dnfile.mdtable.TypeDefRow)):
+        return
+
+    yield Class(DnClass.format_name(row.Class.row.TypeNamespace, row.Class.row.TypeName)), insn.offset
```

Comment:
this looks a little funny, not sure if this is better though?
```suggestion
    yield Class(str(DnClass(row.Class.row.TypeNamespace, row.Class.row.TypeName)), insn.offset
```

Reply:
note that there's other instances like this, if you decide to change it.

Reply:
I'd like to avoid instantiating an object just to format the name. Let's stick with the static method for now.